### ISSUE TITLE
linkcheck should be performed on a merge of master into PR

### ISSUE THREAD

**Issue by DickJC123** [apache/incubator-mxnet#20888]
## Description
**When the unittests are run, the built repo under test is not the PR's HEAD commit, it's a merge of master into the PR.**  One nice thing about this is that if the PR is accepted, the result of merging the accepted PR into master is exactly what was tested.  Also, if a different PR is merged into master that fixes test failures, it immediately helps all PR's in development- they don't need to merge the new master into the PR to pick up the fix.  Sadly, there's a dark side to this: an accepted PR that introduces a flakey test will hamper in-development PR's immediately.

We recently had a linkcheck failure fixed on master by PR https://github.com/apache/incubator-mxnet/pull/20879, but the linkcheck failure was not eliminated for in-development PRs.  Only after remerging master into the PR did the failure disappear. **Shouldn't the same approach of merging master into the PR be used with linkcheck, as it is with the unittests?**
